packages/pwa/app/commerce-api/hooks/useCustomerProductLists.js

```diff
@@ -4,361 +4,374 @@
-                    try {
-                        const productList = self.getProductListPerType(listType)
-                        const productListItem = productList?.customerProductListItems.find(
-                            ({productId}) => productId === event.item.id
-                        )
-                        // if the item is already in the wishlist
-                        // only update the quantity
-                        if (productListItem) {
-                            await self.updateCustomerProductListItem(productList, {
-                                ...productListItem,
-                                quantity: event.item.quantity + productListItem.quantity
-                            })
-                            eventHandler(event)
-                        } else {
-                            await self.createCustomerProductListItem(productList, {
-                                productId: event.item.id,
-                                priority: 1,
-                                quantity: parseInt(event.item.quantity),
-                                public: false,
-                                type: 'product'
-                            })
-                            eventHandler(event)
-                        }
-                    } catch (error) {
-                        errorHandler(error)
-                    }
-                    break
-                }
-
-                case CustomerProductListEventQueue.eventTypes.REMOVE:
-                    try {
-                        await self.deleteCustomerProductListItem(list, item)
-                        eventHandler(event)
-                    } catch (error) {
-                        errorHandler(error)
-                    }
-                    break
-            }
-        })
-    }, [customerProductLists])
+    const {state, actions} = useContext(CustomerProductListsContext)
 
     const self = useMemo(() => {
         return {
-            ...customerProductLists,
-            get showLoader() {
-                return isLoading
-            },
+            data: state.productLists,
 
-            get loaded() {
-                return customerProductLists?.data?.length
+            get isInitialized() {
+                return state.productLists !== undefined
             },
 
-            getProductListPerType(type) {
-                return customerProductLists?.data.find((list) => list.type === type)
+            reset() {
+                actions.reset()
             },
 
-            clearProductLists() {
-                // clear out the product lists in context
-                setCustomerProductLists({})
+            // methods that purely send network requests
+            // these methods does not alter states stored
+            // in the context
+            requests: {
+                /**
+                 * Get customer's product lists.
+                 */
+                getLists: handleAsyncError(() => {
```

Comment:
The function you are passing into `handleAsyncError` are not asynchronous, I'm not sure what kind of errors this is causing, but it's probably not functioning correctly as you are awaiting on a sync function, but it will execute immediately because the api call is async. You probably want to have `async` declared before the function parentheses.

Reply:
sorry my jsdoc comment is misleading and I corrected it.

we are good as long as the `func` is a function and it returns a promise, it doesn't have to use the syntactic suger `async`.